pages/api/token/temporary.ts

```diff
@@ -0,0 +1,75 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import validate, {
+  BearerTokenPayload,
+  createTemporaryToken,
+  updateTemporaryToken,
+} from "@lib/middleware/bearerToken";
+import dbConnector from "@lib/integration/dbConnector";
+import executeQuery from "@lib/integration/queryManager";
+import { formUser } from "@lib/types-database";
+import { QueryResult } from "pg";
+import { logMessage } from "@lib/logger";
+
+const checkRequestPayload = (
+  handler: (
+    req: NextApiRequest,
+    res: NextApiResponse,
+    bearerTokenPayload: BearerTokenPayload,
+    email: string
+  ) => void
+) => {
+  return async (req: NextApiRequest, res: NextApiResponse, options?: unknown): Promise<unknown> => {
+    const requestBody = JSON.parse(req.body);
+    try {
+      if (Object.prototype.hasOwnProperty.call(requestBody, "email")) {
+        return handler(req, res, options as BearerTokenPayload, requestBody["email"]);
+      } else {
+        res.status(400).json({ error: "Invalid payload" });
+      }
+    } catch (err) {
+      res.status(400).json({ error: "Invalid payload" });
+    }
+  };
+};
+
+const handler = async (
+  req: NextApiRequest,
+  res: NextApiResponse,
+  bearerTokenPayload: BearerTokenPayload,
+  email: string
+) => {
+  try {
+    // TODO: create temporary token
+    const formID = bearerTokenPayload.formID;
+    if (formID == undefined) {
+      res.status(403).json({ error: "Invalid form request." });
+      return;
+    }
+    const form = await getForm(formID, email);
```

Comment:
You're doing an extra query for no reason. If the form doesn't exist the UPDATE query will just return a count of 0.